src/DataTable/utils/useConvertIndeterminateProp.js

```diff
@@ -0,0 +1,23 @@
+import { useMemo } from 'react';
+
+/**
+ * A React hook to convert ``indeterminate`` prop to ``isIndeterminate``.
+ *
+ * @param {object} props An object containing props.
+ * @returns Updated props object.
+ */
+const useConvertIndeterminateProp = (props) => {
+  const updatedProps = useMemo(
+    () => {
+      const newProps = { ...props };
+      newProps.isIndeterminate = newProps.indeterminate;
```

Comment:
is reason why not use: `{ ...props, isIndeterminate: props.indeterminate }`?

Reply:
It'd have to be written as the following, such that `indeterminate` is no longer part of `props`, but yes... this is likely cleaner:

```jsx
const { indeterminate, ...rest } = props;
const newProps = { ...rest, isIndeterminate: indeterminate };
return newProps;
```